### PR TITLE
Surface file info as a dict instead of a tuple

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** surface file info as a dict instead of a tuple.

**Benefit:** Treating file info as a dictionary allows us to remove or extend file data more easily when surfacing to Django templates.

**What changed?**
- Change file API to pass file info as a dict instead of a tuple: `file_descriptor_tuples` -> `files_metadata`
- Remove deprecated `FileDescriptor` and `TeamFileDescriptor` named tuples
- Add a Django template filter, `get_item`, to allow unpacking data from a dictionary
- Modify templates to unpack and use dictionary data for file info

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [ ] Test suites passing on Jenkins
- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [JIRA-XXXX](https://openedx.atlassian.net/browse/JIRA-XXXX)

FIY: @edx/masters-devs-gta

### CODE REVIEW

**Comment by jansenk** [comment]
I don't think we particularly want to avoid using the namedtuple in code, just in the template. I think you can call _asdict() on a namedtuple and it returns a dict representation.

File: openassessment/templates/openassessmentblock/oa_team_uploaded_files.html [hunk 1]
```diff
@@ -10,7 +11,9 @@ <h5 class="submission__answer__part__text__title">
         {% endif %}
 
         <div class="submission__answer__files">
-        {% for file_url, file_description, file_name, owner_username in team_file_urls %}
+        {% for file in team_file_urls %}
+        {% with file_url=file|get_item:"download_url" file_description=file|get_item:"description" file_name=file|get_item:"name" owner_username=file|get_item:"uploaded_by" %}
```
Comment (muselesscreator) [inline openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:15]: This is hard to read... is there any way to break it up or put this query into js?

Comment (jansenk) [inline openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:15]: I don't think we need this?
Does `file.download_url`, `file. description` etc not work?

Comment (jansenk) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:15]: actually .... i think that same exact syntax would work for the namedtuples as well

Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:15]: Not that I'm aware of... I can break it up more so its not all on one line and just use the filters where the variable is used, but this makes the change-set easier to grok. It'd basically be the same thing but spread out.

Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:15]: I stand corrected. @jansenk's syntax works and removes the need for a filter.

Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:15]: Yup, this works. Will update.

**Comment by nsprenkle** [comment]
> I don't think we particularly want to avoid using the namedtuple in code, just in the template. I think you can call _asdict() on a namedtuple and it returns a dict representation.

Is there a reson to prefer a `namedtuple` over a `dict` or is that just to reduce the change-set throughout the code?

File: openassessment/templates/openassessmentblock/oa_team_uploaded_files.html [hunk 2]
```diff
@@ -10,25 +10,25 @@ <h5 class="submission__answer__part__text__title">
...
 
         <div class="submission__answer__files">
-        {% for file_url, file_description, file_name, owner_username in team_file_urls %}
-            <div class="submission__answer__team__file__block submission__answer__team__file__block__{{ forloop.counter0 }}" {% if not file_url %} deleted {% endif %}>
-            {% if file_url %}
+        {% for file in team_file_urls %}
+            <div class="submission__answer__team__file__block submission__answer__team__file__block__{{ forloop.counter0 }}" {% if not file.download_url %} deleted {% endif %}>
+            {% if file.download_url %}
                 {% if file_upload_type == "image" %}
                     {% if file_description %}
```
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html:17]: Oof, I'm sort of concerned that the test suite didn't catch this...

[the remark this answers is not in the document]
Comment (jansenk) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html]: do we have a team render_submission test? Might b worthwhile to toss one in there

[the remark this answers is not in the document]
Comment (muselesscreator) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html]: I am adding one for my stuff

[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/templates/openassessmentblock/oa_team_uploaded_files.html]: @muselesscreator, worth me making one or would it be duplication of work?
